galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerContainerResourceMonitor.java

```diff
@@ -0,0 +1,163 @@
+                if (resp.getStatusCode() != 204) {
+                    logger.error("Something went wrong when removing container: " + resp.getStatusLine());
+                }
+            }
+        } catch (HttpClientException e) {
+            logger.error("Failed to kill containers.", e);
+        }        
+    }
+
+    /**
+     * Looks at all containers on a Engine, locates Galasa specifics and ensures they have a decicated slot
+     * @param engine
+     * @param client
+     * @return
+     */
+    private List<String> getOrphanedContainers(String engine, IHttpClient client) {
+        List<String> orphanedContainers = new ArrayList<>();
+        try {
+            HttpClientResponse<String> resp = client.getText("/containers/json");
+            if (resp.getStatusCode() != 200) {
+                logger.error("Something went wrong when retrieving containers: " + resp.getStatusLine());
+            }
+
+            DockerContainerJSON[] activeContainers = gson.fromJson(resp.getContent(), DockerContainerJSON[].class);
+            for (DockerContainerJSON container : activeContainers) {
+                String runName = parseRunId(container.getNames());
+                // Other non Galasa pod
+                if (runName == null) {
+                    continue;
+                }
+
+                Collection<String> activeRunNames = dss.getPrefix("engine."+engine+".slot").values();
+                if (!activeRunNames.contains(runName)) {
+                    orphanedContainers.add(container.getId());
+                }
+            }
+        } catch (HttpClientException | FrameworkException e) {
+            logger.error("Failed to get containers.", e);
+        }
+        return orphanedContainers;
+    }
+
+    /**
+     * Looks at CPS for the docker Engines available
+     * @return
+     */
+    private Map<String, IHttpClient> getDockerEngines() {
+        HashMap<String, IHttpClient> engines = new HashMap<>();
+        try {
+            String[] enginesTags = cps.getPrefixedProperties("engines").get("engines").split(",");
+            for (String engine : enginesTags) {
+                String hostname = cps.getProperty("engine", "hostname", engine);
+                String port = cps.getProperty("engine", "port", engine);
+
+                IHttpClient client = StandAloneHttpClient.getHttpClient(3600, logger);
+                client.setURI(new URI(hostname+":"+port));
+
+                engines.put(engine, client);
+            }
+        } catch (ConfigurationPropertyStoreException | URISyntaxException e) {
+            logger.error("Failed to get docker engines.", e);
+        }
+        return engines;
+    }
+
+    /**
+     * Locates the Run ID from the container name.
+     * 
+     * Name is always structure like "GASLAS_RUNID_CONTAINERTAG"
+     * @param names
+     * @return
+     */
+    private String parseRunId(String[] names) {
```

Comment:
Dont we store the RunID as a label in the container.   Should use that instead

Reply:
Also,  the label should contain the SLOT id,  this would make it easier to determine if the slot is still active, in the above code